topics/icrc_25_signer_interaction_standard.md

```diff
@@ -356,7 +362,11 @@ While processing the request from the relying party, the signer can cancel it at
     - If the user has previously selected identities for the same relying party on the same session, the signer may skip the user interaction.
 4. The signer signs the challenge with the private keys associated with the selected identities and sends a successful response back to the relying party with the list of selected identities and the matching signatures.
 5. After receiving a successful response, the relying party verifies that the signer has access to the private key associated with the provided identities:
-    - The relying party retrieves the `publicKey` from each `identities` value, determines the `signature` scheme and verifies whether it was generated by signing the concatenation of the domain separator `\x13ic-signer-challenge` (UTF-8 encoded) and the `challenge` from the request with the private key associated with the `publicKey`.
+    - If `delegations` is present and non-empty:
+      - The relying party validates that all delegations are not expired according to their respective `expiration` timestamp.
+      - The relying party validates all `signature` values of all delegations in the `delegations` list. For the first delegation in the array, this signature must be valid with respect to the public key corresponding to the `publicKey` field of the identity, all subsequent delegations are signed with the key corresponding to the public key contained in the preceding delegation. The key to validate the challenge signature with is the public key contained in the last delegation in the array.
+    - Otherwise the key to validate the challenge signature with is the public key contained in the `publicKey` field of the identity.
+    - The relying party retrieves the public key to verify the challenge from each identity according to the previous step, determines the `signature` scheme and verifies whether it was generated by signing the concatenation of the domain separator `\x13ic-signer-challenge` (UTF-8 encoded) and the `challenge` from the request with the private key associated with that public key.
```

Comment:
"according to the previous step" is somewhat misleading, as the method of getting the public key for the identity is actually described in the two previous steps.  Maybe restructure the description to merge the previous two steps into one, e.g. something like:
 - determine the public key for verifying the signature of the challenge for each identity: 
    - if there is no delegation (or delegation chain?), use `publicKey`
    - otherwise, verify the delegation chain, and use the public key from the last element of the chain

Reply:
Agreed, fixed.